### DISCUSSION TITLE
Custom validation and required rule

### DISCUSSION THREAD

**Discussion by superandrew** [react-hook-form/react-hook-form#2606] [Q&A] [answered]
I'm pretty new to this awesome form but I have used in a simple react app. I'm having some issues making it work on react native, specifically, I am able to see an error message when the field is required, however if I put a custom validate object on the rules property, the required never shows a message (tried with many setups).

Example:

```
<Controller
              control={control}
              render={({ onChange, onBlur, value }) => (
                <RowInput
                  {...props}
                  label="Codice Fiscale/P.IVA"
                  style={styles.input}
                  editable={editable}
                  keyboardType="default"
                  onChangeText={(value) => onChange(value)}
                  value={value}
                />
              )}
              name="vat_id_or_cf"
              rules={{
                required: true, //this works if no validation is applied
                validate: {
                  1: (value) => value.length > 0, // here I tried also a (value)=>Validation.required(value) which returns a string if it is empty
                  2: (value) => Validation.isValidCFOrPI(value),
                },
              }}
              defaultValue=""
            />
            {errors.vat_id_or_cf && (
              <Text>
                <ErrorMessage errors={errors} name="vat_id_or_cf" /> //should I look for a specific field here?
              </Text>
            )}
```
The only error I see is when the field is populated with wrong values that fail `Validation.isValidCFOrPI` check.

Thank you in advance for your help!

**Comment by bluebill1049** [comment]
can you try to reproduce this in a codesandbox? (we share the same logic as web and RN)

**Comment by superandrew** [comment]
Sure, It's not 100% the same but roughly it shows the problem

https://codesandbox.io/s/trusting-cohen-n5un7?file=/src/App.js

Just focus the field and then blur. Then write something on the field and then blur.

**Reply by bluebill1049** [reply]
`criteriaMode: 'all'`?

**Reply by superandrew** [reply]
I tried but it seems the `required:true` rule doesn't show any message when failed

**Comment by superandrew** [comment, marked as answer]
I was able to fix it.. the `required` function returning a text. However it must be noted that the `required:true` rule doesn't seem to work

**Comment by superandrew** [comment]
It works, anyway. The custom validation required rule was wrong, but I'm not sure if the "required:true" is supposed to work even with custom validation.

**Comment by superandrew** [comment]
For some reason the same code that works con codesandbox doesn't work on react native. I tried just the required function alone but it doesn't show any message

**Comment by superandrew** [comment]
This is my actual RN code:

```<Controller
              control={control}
              render={({ onChange, onBlur, value }) => (
                <RowInput
                  {...props}
                  label="Codice Fiscale/P.IVA"
                  style={styles.input}
                  editable={editable}
                  keyboardType="default"
                  onChangeText={(value) => onChange(value)}
                  value={value}
                />
              )}
              name="vat_id_or_cf"
              rules={{
                validate: {
                  required,
                },
              }}
              defaultValue=""
            />
            {errors.vat_id_or_cf && (
              <Text>
                <ErrorMessage errors={errors} name="vat_id_or_cf" />
              </Text>
            )}
```
There is just the "required" rule that works on react.
